src/private/authnPrivateHandler.ts

```diff
@@ -0,0 +1,70 @@
+import { headers, Subscription } from "nats";
+import AuthnCommand from "../commands/authn/authnCommand";
+import CreateSessionCommand from "../commands/session/createSessionCommand";
+import { HANDLERS_SUBJECTS } from "../config";
+import { authnService, sessionService } from "../di.config";
+import { deserialize, jsonCodec, PrivateNatsHandler } from "../nats/nats";
+import { ConnectorResponse } from "../network/config/connector";
+import { CreateSessionResponse } from "./sessionPrivateHandler";
+
+interface AuthnResponseInterface extends ConnectorResponse {
+    data: {
+        sessionToken: string;
+    };
+}
+
+export const authnPrivateHandlers: PrivateNatsHandler[] = [
+    [
+        HANDLERS_SUBJECTS.AUTHN,
+        async (subscription: Subscription): Promise<void> => {
+            for await (const message of subscription) {
+                try {
+                    const data = deserialize<AuthnCommand>(
+                        message.data,
+                        AuthnCommand
+                    );
+                    const responseHeaders = headers();
+
+                    const authnCommand = new AuthnCommand(data);
+                    const authnResponse = (await authnService.authn(
+                        authnCommand
+                    )) as AuthnResponseInterface;
```

Comment:
I think `authnService.authn()` should return a Promise of `AuthnResponseInterface` instead of something generic that we need to type cast. `ConnectorResponse` should be a type used internally in the services

Reply:
Indeed yes !